enterprise/backend/src/metabase_enterprise/audit_db.clj

```diff
@@ -186,24 +186,53 @@
   :audit      :never
   :doc        false)
 
+(defsetting last-analytics-checksum
+  "Whether or not we should load Metabase analytics content on startup. Defaults to true, but can be disabled via environment variable."
+  :type       :integer
+  :default    0
+  :visibility :internal
+  :audit      :never
+  :doc        false
+  :export?    false)
+
+(defn- analytics-checksum []
+  (reduce
+   (fn [acc file]
+     (+ acc (hash (slurp file))))
+   0
+   (->> (io/file analytics-dir-resource)
+        file-seq
+        (remove fs/directory?))))
+
+(defn- should-load-audit? [analytics-dir-resource load-analytics-content? current-checksum last-checksum]
+  (boolean (and analytics-dir-resource
+                load-analytics-content?
+                (not= current-checksum last-checksum))))
+
 (defn- maybe-load-analytics-content!
   [audit-db]
-  (when (and analytics-dir-resource (load-analytics-content))
-    (ee.internal-user/ensure-internal-user-exists!)
-    (adjust-audit-db-to-source! audit-db)
-    (log/info "Loading Analytics Content...")
-    (ia-content->plugins (plugins/plugins-dir))
-    (log/info (str "Loading Analytics Content from: " (instance-analytics-plugin-dir (plugins/plugins-dir))))
-    ;; The EE token might not have :serialization enabled, but audit features should still be able to use it.
-    (let [report (log/with-no-logs
-                   (serialization.cmd/v2-load-internal! (str (instance-analytics-plugin-dir (plugins/plugins-dir)))
-                                                        {:backfill? false}
-                                                        :token-check? false))]
-      (if (not-empty (:errors report))
-        (log/info (str "Error Loading Analytics Content: " (pr-str report)))
-        (log/info (str "Loading Analytics Content Complete (" (count (:seen report)) ") entities loaded."))))
-    (when-let [audit-db (t2/select-one :model/Database :is_audit true)]
-      (adjust-audit-db-to-host! audit-db))))
+  (let [current-checksum (analytics-checksum)
+        last-checksum (last-analytics-checksum)]
```

Comment:
maybe move these into `should-load-audit?`?

Reply:
I pulled this out to make it easier to test o:)